uplink/decorators.py

```diff
@@ -103,10 +103,13 @@ class headers(MethodAnnotation):
     """
 
     def __init__(self, arg, **kwargs):
-        # TODO: allow the first argument to be a list or str, in
-        # which case you would split the strings by a colon delimiter
-        # to identify the headers.
-        self._headers = dict(arg, **kwargs)
+        if isinstance(arg, str):
+            header = arg.replace(":", "").split(" ")
```

Comment:
Here, I think you can instead split by `:` then strip both the key and value. For instance, that would allow for the cases like `"key:value"`.
```
key, value = arg.split(":")
key = key.strip()
value = value.strip()
```

Reply:
Or, a one liner:

```python
key, value = map(str.strip, arg.split(":"))
```